hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -114,17 +129,38 @@ public class AbfsOutputStream extends OutputStream implements Syncable,
   private final DataBlocks.BlockFactory blockFactory;
 
   /** Current data block. Null means none currently active. */
-  private DataBlocks.DataBlock activeBlock;
+  private AbfsBlock activeBlock;
 
   /** Count of blocks uploaded. */
   private long blockCount = 0;
 
   /** The size of a single block. */
   private final int blockSize;
 
+  /** The map to store blockId and Status **/
+  private final LinkedHashMap<String, BlockStatus> map = new LinkedHashMap<>();
+
+  /** The list of already committed blocks is stored in this list. */
+  private List<String> committedBlockEntries = new ArrayList<>();
+
+  /** The list of all blockId's for putBlockList. */
+  private final Set<String> blockIdList = new LinkedHashSet<>();
+
+  /** The prefix mode for decision on BLOB or DFS endpoint. */
+  private PrefixMode prefixMode;
+
+  /** The etag of the blob. */
+  private String eTag;
+
   /** Executor service to carry out the parallel upload requests. */
   private final ListeningExecutorService executorService;
 
+  /** List to validate order. */
+  private final UniqueArrayList<String> orderedBlockList = new UniqueArrayList<>();
+
+  /** Retry fallback for append on DFS */
+  private int retryAppendDFS = 0;
```

Comment:
maybe rename to fallbackDFSAppendCount

Reply:
Recording what we discussed:
We will add metric for when the file falls back to DFS int racking context. One slot will be placed with default "B" which will change to D when fallback happens. useragent + this endpoint indicator will tell us if fallback happened from v2.